### PR TITLE
chore: Update datafusion

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
1. Get fix for https://github.com/apache/arrow-datafusion/pull/1047 (FYI @NGA-TRAN)

# Changes:
1. Update to use latest datafusion
2. run `cargo update` and check in results
2. Update to latest sqlparser, remove unused dependencies
3. Enable tests that were failing due to https://github.com/apache/arrow-datafusion/pull/1047 (changes in https://github.com/influxdata/influxdb_iox/pull/2618)
4. Update explain plan output due improvements in display from  https://github.com/apache/arrow-datafusion/pull/971  @matthewmturner 

# Other crates that are updated:

```
alamb@ip-192-168-182-158 influxdb_iox % cargo update
...
cargo update
    Updating crates.io index
    Updating git repository `https://github.com/apache/arrow-datafusion.git`
    Updating git repository `https://github.com/Azure/azure-sdk-for-rust.git`
    Updating git repository `https://github.com/mkmik/heappy`
    Updating instant v0.1.10 -> v0.1.11
    Updating openssl-sys v0.9.66 -> v0.9.67
    Updating pkg-config v0.3.19 -> v0.3.20
    Updating syn v1.0.76 -> v1.0.77
    Updating tinyvec v1.4.0 -> v1.5.0
    Updating zeroize v1.4.1 -> v1.4.2
```

### CODE REVIEW

File: query/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,6 @@ observability_deps = { path = "../observability_deps" }
 parking_lot = "0.11.2"
 regex = "1"
 snafu = "0.6.9"
-sqlparser = "0.10.0"
```
Comment (alamb) [inline query/Cargo.toml:32]: unused

**Review by e-dard — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -142,9 +142,6 @@ jobs:
       # "1" means line tables only, which is useful for panic tracebacks.
       RUSTFLAGS: "-C debuginfo=1"
       RUST_BACKTRACE: "1"
-      # set min stack size as a workaround to avoid stack overflow bug in DF
-      # https://github.com/apache/arrow-datafusion/issues/419
-      RUST_MIN_STACK: "10485760"
```
Comment (NGA-TRAN) [inline .circleci/config.yml:147]: 🎉

File: datafusion_util/src/lib.rs [hunk 3]
```diff
@@ -209,7 +209,7 @@ mod tests {
 
         let ts_predicate_expr = make_range_expr(101, 202, "time");
         let expected_string =
-            "TimestampNanosecond(101) LtEq #time And #time Lt TimestampNanosecond(202)";
+            "TimestampNanosecond(101) <= #time AND #time < TimestampNanosecond(202)";
```
Comment (NGA-TRAN) [inline datafusion_util/src/lib.rs:212]: Nice

File: query_tests/cases/in/pushdown.expected [hunk 4]
```diff
@@ -10,158 +10,158 @@
...
+| logical_plan  | Projection: #restaurant.count, #restaurant.system, #restaurant.time, #restaurant.town                       |
+|               |   Filter: #restaurant.count > Int64(200)                                                                    |
+|               |     TableScan: restaurant projection=Some([0, 1, 2, 3]), filters=[#restaurant.count > Int64(200)]           |
+| physical_plan | ProjectionExec: expr=[count@0 as count, system@1 as system, time@2 as time, town@3 as town]                 |
+|               |   CoalesceBatchesExec: target_batch_size=500                                                                |
+|               |     FilterExec: CAST(count@0 AS Int64) > 200                                                                |
+|               |       RepartitionExec: partitioning=RoundRobinBatch(4)                                                      |
+|               |         IOxReadFilterNode: table_name=restaurant, chunks=1 predicate=Predicate exprs: [#count > Int64(200)] |
++---------------+-------------------------------------------------------------------------------------------------------------+
```
Comment (NGA-TRAN) [inline query_tests/cases/in/pushdown.expected:24]: I like the changes that easier to read

**Review by NGA-TRAN — APPROVED** [review]
Looks great.

File: query_tests/cases/in/pushdown.expected (hunk 4 above)
Comment (alamb) [reply query_tests/cases/in/pushdown.expected:24]: 👍  me too -- that is due to https://github.com/apache/arrow-datafusion/pull/971 from @matthewmturner  ❤️

**Pull request merged**